connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -291,6 +291,7 @@ public DistributedHerder(DistributedConfig config,
 
         String clientIdConfig = config.getString(CommonClientConfigs.CLIENT_ID_CONFIG);
         String clientId = clientIdConfig.length() <= 0 ? "connect-" + CONNECT_CLIENT_ID_SEQUENCE.getAndIncrement() : clientIdConfig;
+        String escapedClientId = clientId.replace("%", "%%");
```

Comment:
This isn't mentioned in the description for the PR or the Jira ticket; why is this changed?

Reply:
This was caught by one of the MM2 tests using a cluster alias with a % inside. That caused the thread factory naming pattern to crash, as String.format thinks % represents some placeholder. Will add a comment here.

Reply:
🙇 Good eye!